[website] Add new legal pages (#33650)
mui/material-ui: docs/pages/blog/premium-plan-release.md

Here is a list of everything that has changed:

:::warning

> ⚠️ Please note: The following **pricing** changes apply only to **new customers**. Existing customers are grandfathered, they can renew their license under the [previous terms](https://mui.com/legal/mui-x-eula-2022-05-08/) once. Their last renewal can have a support duration of up to 5 years.

Reviewer: ```suggestion
⚠️ Please note: The following **pricing** changes apply only to **new customers**. Existing customers are grandfathered in, they can renew their license under the [previous terms](https://mui.com/legal/mui-x-eula-2022-05-08/).
```

This was an idea you floated, but it was never agreed. To be removed ASAP.

Author: Interesting, I think that we should consider this for a follow-up. This PR is about creating a new legal scope, so it's off-topic.

I feel that almost nobody renews, so no objections to moving in the direction you are pushing here. Personally, I would be in favor of keeping the current tradeoffs so we can more quickly "remove pricing debt".

>   The price for Pro is still capped at 10 seats, but we’re removing the concept of price cap for Premium. We kindly ask you to [contact sales](mailto:sales@mui.com) if you wish to purchase a license for more than 50 developers.

Reviewer: ```suggestion
  Pro is still capped at 10 seats, but Premium doesn't have a license cap. We kindly ask you to [contact sales](mailto:sales@mui.com) if you wish to purchase a license for more than 50 developers.
```

> Please, feel free to read the new [license agreement](https://mui.com/legal/mui-x-eula/) in detail.

Reviewer: ```suggestion
Please feel free to read the [license agreement](https://mui.com/legal/mui-x-eula/) in detail.
```
